pkg/hcl2nix/lock.go

```diff
@@ -97,8 +97,18 @@ func GenerateLockFile(conf *Config, packages []LockPackage, wr io.Writer) error
 }
 
 // ResolvePackages resolves a list of packages concurrently
-func ResolvePackages(ctx context.Context, sc buildsafev1.SearchServiceClient, packages Packages) ([]LockPackage, error) {
-	allPackages := slices.Compact(append(packages.Development, packages.Runtime...))
+func ResolvePackages(ctx context.Context, sc buildsafev1.SearchServiceClient, packages Packages, pkgType string) ([]LockPackage, error) {
+	var selectedPackages []string
+    switch pkgType {
+    case "dev":
+        selectedPackages = packages.Development
+    case "runtime":
+        selectedPackages = packages.Runtime
+    case "all":
```

Comment:
when they select `dev`, we want to return `all` . 
Think of runtime being a subset of dev

Reply:
you mean if they select dev, we generate it for dev and runtime both?

Reply:
yes